syntax = "proto3";

package xray.app.stats.command;

message GetStatsRequest {
  string name = 1;
  bool reset = 2;
}

message Stat {
  string name = 1;
  int64 value = 2;
}

message GetStatsResponse {
  Stat stat = 1;
}

message QueryStatsRequest {
  string pattern = 1;
  bool reset = 2;
}

message QueryStatsResponse {
  repeated Stat stat = 1;
}

message GetStatsOnlineIpListResponse {
  string name = 1;
  map<string, int64> ips = 2;
}

service StatsService {
  rpc GetStats(GetStatsRequest) returns (GetStatsResponse);
  rpc GetStatsOnline(GetStatsRequest) returns (GetStatsResponse);
  rpc QueryStats(QueryStatsRequest) returns (QueryStatsResponse);
  rpc GetStatsOnlineIpList(GetStatsRequest) returns (GetStatsOnlineIpListResponse);
}
